pkg/sdkserver/sdkserver_test.go

```diff
@@ -745,6 +753,14 @@ func TestSDKServerWatchGameServerFeatureSDKWatchSendOnExecute(t *testing.T) {
 
 func TestSDKServerSendGameServerUpdate(t *testing.T) {
 	t.Parallel()
+
+	agruntime.FeatureTestMutex.Lock()
```

Comment:
Happy to push this through to remove the flakiness, but curious why we need these here for these tests?

I couldn't see anything specific in here that required the lock?

Would be good to have a note somewhere indicating where the deadlock was generated, so we can avoid it in future tests.

Reply:
Down the lines we are setting FeatureSDKWatchSendOnExecute to false. Without this lock some other test could set it back to true while this test got executed.
I can check this by setting next to true:
```
err := agruntime.ParseFeatures(string(agruntime.FeatureSDKWatchSendOnExecute) + "=false")
```

Reply:
I guess my point was, if it did flip to true, does the test actually fail?

For the sake of expediency, happy for it to stay for now - I was just wondering.

Reply:
If I set "=true" here (or some other parallel test would do that) then test would fail:
```
--- FAIL: TestSDKServerUpdateEventHandler (19.22s)
    sdkserver_test.go:1299:
                Error Trace:    sdkserver_test.go:1299
                                                        asm_amd64.s:1357
                Error:          Expected nil, but got: could not retrieve GameServer default/test: gameserver.agones.dev "test" not found
                Test:           TestSDKServerUpdateEventHandler
    sdkserver_test.go:817:
                Error Trace:    sdkserver_test.go:817
                Error:          Expected nil, but got: &errors.errorString{s:"timed out waiting for
the condition"}
```

Reply:
Got it! Thanks! Then yes, let's definitely keep the lock there! :+1: